issue_1/contracts/WordRegistry.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.4;
+
+import "hardhat/console.sol";
+
+import "./yield-utils-v2/contracts/access/Ownable.sol";
+
+/**
+@title Dictatorship-friendly word registry
+@author Owleksiy
+@notice This contract allows users to register/unregister (claim/unclaim) words
+Once a word is regiistered by a user, other users can't register it, unless the
+original user unregisters it.
+
+Additionally, the owner of the contract can expropriate any registered word
+ */
+contract WordRegistry is Ownable {
+    event OwnershipChange(string word, address previousOwner, address newOwner);
+    // word -> owner mapping
+    mapping(string => address) private word_to_owner;
+
+    constructor() {
+        console.log("Creating a registry, owner:", msg.sender);
+    }
+
+    function _is_owned_by_somebody(string memory word) private view returns (bool) {
```

Comment:
You could have done this with a modifier and it would be a bit clearer.

However, given that this function is called only once and is only one line, I would do this check inline in `register`.

Reply:
100% ack the 2nd point
80% ack the 1st one (I do feel modifiers are actually easy to miss when reading the code), but I'll follow your lead and make the change